core/src/tile/tileManager.cpp

```diff
@@ -239,6 +254,14 @@ void TileManager::clearTileSets(bool clearSourceCaches) {
 void TileManager::clearTileSet(int32_t _sourceId) {
     for (auto& tileSet : m_tileSets) {
         if (tileSet.source->id() != _sourceId) { continue; }
+
+        for (auto& tile : tileSet.tiles) {
+            auto& entry = tile.second;
+            if (entry.isInProgress()) {
```

Comment:
Instead of duplicating the code here, directly use `cancelTasks` as done in `clearTileSets`.

Reply:
Right - I've added this commit with this cleanup to platform-work PR already. Is that ok or do you prefer to merge them separately?

Reply:
Nopes thats good. Already checked that. Thanks.